From 46be1a67acf79ec457dcbb214f258d7fcddeeafa Mon Sep 17 00:00:00 2001
From: Paul Sajna <sajattack@postmarketos.org>
Date: Wed, 3 Jun 2026 21:29:46 -0700
Subject: [PATCH 128/211] dt-bindings: input: touchscreen: Add docs for
 siw,sw49410-ts-spi

Document devicetree usage for SiliconWorks SW49410 Touchscreen

Signed-off-by: Paul Sajna <sajattack@postmarketos.org>
---
 .../input/touchscreen/siw,sw49410-ts-spi.yaml | 59 +++++++++++++++++++
 1 file changed, 59 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/input/touchscreen/siw,sw49410-ts-spi.yaml

diff --git a/Documentation/devicetree/bindings/input/touchscreen/siw,sw49410-ts-spi.yaml b/Documentation/devicetree/bindings/input/touchscreen/siw,sw49410-ts-spi.yaml
new file mode 100644
index 000000000000..d541c35a6ab6
--- /dev/null
+++ b/Documentation/devicetree/bindings/input/touchscreen/siw,sw49410-ts-spi.yaml
@@ -0,0 +1,59 @@
+# SPDX-License-Identifier: (GPL-2.0 OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/input/touchscreen/siw,sw49410-ts-spi.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: SiliconWorks SW49410 SPI touchscreen
+
+maintainers:
+  - Paul Sajna <sajattack@postmarketos.org>
+
+description: SiliconWorks touchscreen connected with SPI
+
+allOf:
+  - $ref: touchscreen.yaml#
+  - $ref: /schemas/spi/spi-peripheral-props.yaml#
+
+properties:
+  compatible:
+    enum:
+      - siw,sw49410-ts-spi
+
+  interrupts:
+    maxItems: 1
+
+  reset-gpios:
+    maxItems: 1
+
+required:
+  - compatible
+  - interrupts
+  - reset-gpios
+  - touchscreen-size-x
+  - touchscreen-size-y
+
+unevaluatedProperties: false
+
+examples:
+  - |
+    #include <dt-bindings/gpio/gpio.h>
+    #include <dt-bindings/interrupt-controller/irq.h>
+
+    &spi5 {
+      clock-frequency = <5000000>;
+
+      status = "okay";
+
+      touchscreen@0 {
+          compatible = "siw,sw49410-ts-spi";
+          reg = <0>;
+          interrupts-extended = <&tlmm 31 IRQ_TYPE_EDGE_FALLING>;
+          spi-max-frequency = <5000000>;
+          reset-gpio = <&tlmm 32 GPIO_ACTIVE_LOW>;
+          panel = <&display_panel>;
+          touchscreen-size-x = <1440>;
+          touchscreen-size-y = <3120>;
+        };
+    };
+...
-- 
2.55.0

